### PR TITLE
Project Elephant Quiz made by Sofie Pellegrini, Lisa Bergström, Jessica Sandler, Terese Claesson and Emma Berg

### PR DESCRIPTION
**Pull request by EmmaaBerg**
https://elephantquiz.netlify.app/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,9 @@
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
     "react-redux": "^7.1.3",
-    "react-scripts": "^3.3.0"
+    "react-router-dom": "^6.3.0",
+    "react-scripts": "^3.3.0",
+    "use-sound": "^4.0.1"
```
Comment (Dopest0727) [inline package.json:19]: Great use of sound effects

File: src/App.js [hunk 2]
```diff
@@ -14,7 +15,14 @@ const store = configureStore({ reducer })
...
     <Provider store={store}>
-      <CurrentQuestion />
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<StartPage></StartPage>}>
+          </Route>
+          <Route path="/quiz" element={<CurrentQuestion></CurrentQuestion>}>
+          </Route>
+        </Routes>
+      </BrowserRouter>
```
Comment (Dopest0727) [inline src/App.js:25]: Good structure of project, easy to maintain

File: src/index.css [hunk 3]
```diff
@@ -1,13 +1,267 @@
...
+    height: 100%;
+    z-index: 1;
+    transition: all 0.3s;
+    background-color: rgba(250, 19, 19, 0.1);
+  }
+  .btn-restart:hover::after {
+    opacity: 0; 
+    transform: scale(0.1, 1);
+  }
+
```
Comment (Dopest0727) [inline src/index.css:266]: Good styling over, we would recommend you to use Styled Components because later on as we start with React Native it will be the only option.

File: src/components/Button.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+export const StartButton = () => {
+
+  const [playsound] = useSound(mySound)
+
+  return ( 
+  <button className="start-button" onClick={playsound}>
+  <span>START QUIZ HERE!</span></button>
+  )
+}
```
Comment (Dopest0727) [inline src/components/Button.js:15]: Good use of the sound whilst pressing the button (Scared us for a second M&R)

File: src/components/Button.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    </button>
+  );
+};
+
+export const NextQuestion = ({ setDisabled }) => {
+  const dispatch = useDispatch();
+  const nextQuestion = () => {
+    dispatch(quiz.actions.goToNextQuestion());
+    setDisabled(false);
+  };
```
Comment (Dopest0727) [inline src/components/Button.js:72]: Validation is not enabled when we played the game.

File: src/components/Footer.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+
+export const Footer = () => {
+    return (
+        <footer>
+            <p>Made by Sofie Pellegrini, Lisa Bergström, Jessica Sandler, Emma Berg and Terese Claesson </p>
+            <p>Technigo Bootcamp Spring 2022 </p>
+        </footer>
+    )
+}
```
Comment (Dopest0727) [inline src/components/Footer.js:10]: Great Job!

File: src/components/CurrentQuestion.js [hunk 7]
```diff
@@ -1,16 +1,52 @@
...
+      </div>
+      <NextQuestion setDisabled={setDisabled} />
+      <p className="question-number"> Question {question.id} / {numberOfQuestions}</p>
     </div>
-  )
-}
+    <Footer />
+    </main>
+  );
+};
```
Comment (Dopest0727) [inline src/components/CurrentQuestion.js:52]: Good overall logic in this component. A lot going on but is easy to read! 👍

File: src/components/Summary.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { useSelector} from 'react-redux'
+import { RestartButton } from './Button'
+import { Footer } from 'components/Footer'
+import { Link } from 'react-router-dom'
```
Comment (Dopest0727) [inline src/components/Summary.js:5]: It is considered good practice to separate imported components/packages in the file with a space.

File: src/components/StartPage.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+export const StartPage = () => {
+
+  return (
+    <main className="startpage">
+      <div className='background'>
+        <h1 className='intro-title'>Test your elephant knowledge here!</h1>
+        <Link to="/quiz">
+          <StartButton/>
+        </Link>
+        <img className="icon" src={icon} alt="icon" />
```
Comment (Dopest0727) [inline src/components/StartPage.js:16]: Good use of "alt" for accessibility reasons

File: src/startpage.css [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    z-index: 1;
+    transition: all 0.3s;
+    background-color: rgba(250, 19, 19, 0.1);
+  }
+  
+  .start-button:hover::after {
+    opacity: 0;
+    transform: scale(0.1, 1);
+  }
+}
```
Comment (Dopest0727) [inline src/startpage.css:105]: Other than the responsiveness of the start page the css is on point guys!

**Review by Dopest0727 — COMMENTED** [review]
Great job, the code is really easy to read and understand. No need for comments in our opinions since all the components/variables were self explanatory. 
We encourage you to start using Styled Components other than that GREAT FIRST REDUX PROJECT! 
From Maurii & Rijad @rijad90

File: src/components/Button.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { quiz } from 'reducers/quiz';
+import mySound  from '../assets/elephantSound.mp3';
+import useSound from 'use-sound';
+
+export const StartButton = () => {
+
+  const [playsound] = useSound(mySound)
```
Comment (Laura-Sjolander) [inline src/components/Button.js:9]: Very surprising with the sound effect! Gets you wide awake and ready to take on the quiz!

File: src/components/CurrentQuestion.js [hunk 12]
```diff
@@ -1,16 +1,52 @@
...
+            index={index}
+            questionId={question.id}
+            question={question}
+            answer={userAnswer}
+            setDisabled={setDisabled}
+            disabled={disabled}
+          />
+        ))}
+      </div>
+      <NextQuestion setDisabled={setDisabled} />
```
Comment (Laura-Sjolander) [inline src/components/CurrentQuestion.js:46]: Nice that you were able to disable the buttons!

File: src/index.css [hunk 13]
```diff
@@ -1,13 +1,267 @@
...
+p {
+  margin: 5px;
+}
+
+.correct {
+  border: 2px solid #4FB477;
+  margin-bottom: 10px;
+  padding: 10px;
+  font-family: 'Raleway', sans-serif;
+}
```
Comment (Laura-Sjolander) [inline src/index.css:41]: Clear and easy to understand with red/green borders, when you get the answer wrong or right.

File: src/index.css [hunk 14]
```diff
@@ -1,13 +1,267 @@
...
+
+/**************TABLET VIEW*******************/
+@media screen and (min-width: 668px) {
+
+  .summary-text{
+    font-size:20px;
+  }
+}
+
+
```
Comment (Laura-Sjolander) [inline src/index.css:155]: Maybe you could put mediaqueries also for the startpage? The backgroundimage looks great in ipad mini, but doesnt stretch out in the other formats.

**Review by Laura-Sjolander — COMMENTED** [review]
Nice work team! Overall a clear and structured code. Nice design and the functionality works great.

File: src/App.js [hunk 15]
```diff
@@ -14,7 +15,14 @@ const store = configureStore({ reducer })
 export const App = () => {
   return (
     <Provider store={store}>
-      <CurrentQuestion />
+      <BrowserRouter>
+        <Routes>
+          <Route path="/" element={<StartPage></StartPage>}>
```
Comment (nadialefebvre) [inline src/App.js:20]: Here you could write only `<StartPage />` when it's a self closing tag.

File: src/components/Button.js (hunk 11 above)
Comment (nadialefebvre) [reply src/components/Button.js:9]: I agree, really interesting!

File: src/components/Button.js [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+export const AnswerButton = ({
+  item,
+  index,
+  questionId,
+  question,
+  answer,
+  setDisabled,
+  disabled,
```
Comment (nadialefebvre) [inline src/components/Button.js:40]: Great that you split it all on many lines, no need to scroll to the right.

File: src/components/Button.js [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+  };
+
+  return (
+    <button
+      className={
+        !answer
+          ? 'defaultbtn'
+          : index === question.correctAnswerIndex
+          ? 'correct'
+          : 'wrong'
```
Comment (nadialefebvre) [inline src/components/Button.js:55]: It makes the code shorter with ternary operator, I like that :)

File: src/components/Summary.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+      return (
+        <>
+      <div className='summary'>
+        <h1>This is your summary</h1>
+        {userAnswer.map((answer) => (
+         <div className='summary-text' key={answer.questionId}>
+              <h5>
+              {' '}
+              {answer.questionId}: Correct answer: {' '}
+              {answer.question.options[answer.question.correctAnswerIndex]}
```
Comment (nadialefebvre) [inline src/components/Summary.js:22]: Maybe it's only github that creates this "issue", but indentation is a bit hard to follow in some places, like here.

File: src/index.css [hunk 19]
```diff
@@ -1,13 +1,267 @@
...
+  margin-left: 1.2rem;
+  margin-right: 1.2rem;
+  padding: 15px;
+}
+
+.defaultbtn {
+  background-color: #D3C4BE;
+  margin-bottom: 10px;
+  padding: 10px;
+  font-family: 'Raleway', sans-serif;
```
Comment (nadialefebvre) [inline src/index.css:82]: I see that you repeat many times the font-family: maybe there would be a way to keep it more DRY and specify it only once?

File: src/reducers/quiz.js [hunk 20]
```diff
@@ -2,8 +2,31 @@ import { createSlice } from '@reduxjs/toolkit'
 
 // Change these to your own questions!
 const questions = [
-  { id: 1, questionText: 'Who set the Olympic record for the 100m dash in 2012?', options: ['Usain Bolt', 'Justin Gatlin', 'Tyson Gay', 'Asafa Powell'], correctAnswerIndex: 0 },
-  { id: 2, questionText: 'When was Michael Phelps last named male World Swimmer of the Year?', options: ['2012', '2014', '2016', '2018'], correctAnswerIndex: 2 }
+  { id: 1, 
+    questionText: 'How many muscles are there in an elephant trunk?',
```
Comment (nadialefebvre) [inline src/reducers/quiz.js:6]: Great to learn so many things about us, elephants!
(not code related at all!)

**Review by nadialefebvre — COMMENTED** [review]
Great project, Emma, Sofia, Lisa, Jessica and Terese! You did a very good job!
